be/src/Project/Services/Identity/Identity.Api/Services/Users/UsersService.cs

```diff
@@ -206,21 +206,27 @@ public async Task<UserServiceModel> SetPasswordAsync(SetUserPasswordServiceModel
 
         public async Task<UserServiceModel> UpdateAsync(UpdateUserServiceModel serviceModel)
         {
-            var existingUser = await this.identityContext.Accounts.FirstOrDefaultAsync(x => x.Id == serviceModel.Id.ToString());
+            var existingUser = await this.identityContext.Accounts.FirstOrDefaultAsync(x => x.Email == serviceModel.Email);
             if (existingUser == null)
             {
                 throw new CustomException(this.accountLocalizer.GetString("UserNotFound"), (int)HttpStatusCode.NotFound);
             }
 
+            var organisation = await this.identityContext.Organisations.FirstOrDefaultAsync(x => x.Id == existingUser.OrganisationId);
+            if (organisation is not null)
```

Comment:
**suggestion:** Instead of it please check if organisation is null and throw an exception if it is. Look at lines 210-212 for reference.